src/newfight.cpp

```diff
@@ -784,7 +784,15 @@ void hit_with_multiweapon_toggle(struct char_data *attacker, struct char_data *v
       SEND_RBUF_TO_ROLLS_FOR_BOTH_ATTACKER_AND_DEFENDER;
       
       combat_message(att->ch, def->ch, att->weapon, -1, att->ranged->burst_count);
-      damage(att->ch, def->ch, -1, att->ranged->dam_type, 0);
+      bool target_died = 0;
+      target_died = damage(att->ch, def->ch, -1, att->ranged->dam_type, 0);
+      
+      //Handle suprise attack/alertness here -- ranged attack failed.
+      if (!target_died && AFF_FLAGGED(def->ch, AFF_SURPRISE))
+        AFF_FLAGS(def->ch).RemoveBit(AFF_SURPRISE);
+           
+      GET_MOBALERT(def->ch) = MALERT_ALARM;
```

Comment:
These should go in a !target_died block as well, otherwise there's a chance they'll operate on invalid characters. Also, they probably should be guarded by `if (IS_NPC(def->ch))` to avoid setting these values on player characters.

Reply:
Of course. Did the updates half-asleep so I missed it heh.